tests/tests/gas_price.rs

```diff
@@ -4,156 +4,276 @@ use crate::helpers::{
+    let srv = FuelService::new_node(node_config.clone()).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+    let mut rng = rand::rngs::StdRng::seed_from_u64(2322u64);
+
+    // when
+    let arb_tx_count = 5;
+    for i in 0..arb_tx_count {
+        let tx = arb_large_tx(189028 + i as Word, &mut rng);
+        let _status = client.submit(&tx).await.unwrap();
+    }
+    // starting gas price
+    let _ = client.produce_blocks(1, None).await.unwrap();
+    // updated gas price
+    let _ = client.produce_blocks(1, None).await.unwrap();
+
+    // then
+    let change = starting_gas_price * percent / 100;
+    let expected = starting_gas_price - change;
+    let latest = client.latest_gas_price().await.unwrap();
+    let actual = latest.gas_price;
+    assert_eq!(expected, actual);
+}
+
+#[tokio::test]
+async fn estimate_gas_price__matches_actual_updated_gas_price() {
+    // given
+    let mut node_config = Config::local_node();
+    let starting_gas_price = 1000;
+    let percent = 10;
+    node_config.starting_gas_price = starting_gas_price;
+    node_config.gas_price_change_percent = percent;
+    // Always increase
+    node_config.gas_price_threshold_percent = 0;
+
     let srv = FuelService::new_node(node_config.clone()).await.unwrap();
     let client = FuelClient::from(srv.bound_address);
 
     // when
     let arbitrary_horizon = 10;
 
-    let EstimateGasPrice { gas_price } =
-        client.estimate_gas_price(arbitrary_horizon).await.unwrap();
+    let estimate = client.estimate_gas_price(arbitrary_horizon).await.unwrap();
+    let _ = client.produce_blocks(1, None).await.unwrap();
+    for _ in 0..arbitrary_horizon {
+        let _ = client.produce_blocks(1, None).await.unwrap();
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
 
     // then
-    let expected = node_config.static_gas_price;
-    let actual = u64::from(gas_price);
+    let latest = client.latest_gas_price().await.unwrap();
+    let real = latest.gas_price;
+    let estimated = u64::from(estimate.gas_price);
+    assert_eq!(estimated, real);
+}
+
+#[tokio::test]
+async fn latest_gas_price__if_node_restarts_gets_latest_value() {
+    // given
+    let args = vec![
+        "--debug",
+        "--poa-instant",
+        "true",
+        "--starting-gas-price",
+        "1000",
+        "--gas-price-change-percent",
+        "10",
+        "--gas-price-threshold-percent",
+        "0",
+    ];
+    let driver = FuelCoreDriver::spawn(&args).await.unwrap();
+    let starting = driver.node.shared.config.starting_gas_price;
+    let arb_blocks_to_produce = 10;
+    for _ in 0..arb_blocks_to_produce {
+        driver.client.produce_blocks(1, None).await.unwrap();
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+    let latest_gas_price = driver.client.latest_gas_price().await.unwrap();
+    let LatestGasPrice { gas_price, .. } = latest_gas_price;
+    let expected = gas_price;
+    assert_ne!(expected, starting);
```

Comment:
should we make this something other than the starting price? To ensure we aren't just defaulting to the starting price on restart.

Reply:
This is `ne` for that reason :) Maybe I'm missing something.

Reply:
ah i missed that